precompiles/bank/types.go

```diff
@@ -0,0 +1,34 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package bank
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	cmn "github.com/evmos/evmos/v15/precompiles/common"
+)
+
+// Balance contains the amount for a corresponding ERC-20 contract address
+type Balance struct {
+	ContractAddress common.Address
+	Amount          *big.Int
+}
+
+// ParseBalances parses the call arguments for the bank Balances query.
+func ParseBalances(args []interface{}) (sdk.AccAddress, error) {
```

Comment:
I would prefer `ParseBalancesArgs` here to be more verbose, sounds like you're actually parsing the balances here